### DISCUSSION TITLE
Selenide + GitLab JUnit attachments

### DISCUSSION THREAD

**Discussion by prafair** [selenide/selenide#1884] [Q&A] [answered]
GitLab has a feature: https://docs.gitlab.com/ee/ci/testing/unit_test_reports.html#view-junit-screenshots-on-gitlab implemented in [this](https://gitlab.com/gitlab-org/gitlab/-/issues/202114) task.

From [issue](https://gitlab.com/gitlab-org/gitlab/-/issues/6061#further-details): 
> This issue aims to implement a specific view for contents of screenshotPath and the corresponding junit failures wether that is coming from webdriver/selenium (the GitLab case), cucumber or cypress, all of which provide junit output.

I didn't find that such a thing was supported in Selenide. Based on the comment above, should Selenide support this out of the box? Thanks for the answer in advance =)

**Comment by prafair** [comment]
@asolntsev can you answer please?)

**Comment by asolntsev** [comment, marked as answer]
@prafair yes, sorry for the delay.
Yes, making tests easy to debug (incl. screenshots) is one of primary goals for selenide. 

So yes, selenide could implement this feature (either in core or a "gitlab plugin").

But  I personally don't have any experience  with gitlab. So feel free to submit a pull request,  we will review it.

**Comment by vtintillier** [comment]
For completeness this `[[ATTACHMENT|path/to/file]]` string is also parsed by this Jenkins plugin: https://plugins.jenkins.io/junit-attachments/

**Reply by asolntsev** [reply]
@vtintillier I like the idea. Yes, let's log the line in `[[ATTACHMENT|/path/to/some/file]]` format. 
Feel free to submit a pull request! ;)
